File: Lab 4/C++/Triangle.cpp
#include "Triangle.h"
#include <cmath>

Triangle::Triangle() : Figure() {
    // Виклик базового конструктора за замовчуванням
}

Triangle::Triangle(double xVal, double yVal, double sideVal)
    : Figure(xVal, yVal, sideVal) {
    // Явний виклик конструктора базового класу
}

Triangle::Triangle(const Triangle& other)
    : Figure(other) {
    // Виклик копіюючого конструктора базового класу
}

double Triangle::area() const {
    // Формула площі рівностороннього трикутника: (sqrt(3)/4)*side^2
    return (std::sqrt(3.0) / 4.0) * side * side;
}

double Triangle::perimeter() const {
    return 3.0 * side;
}

File: Lab 4/C#/Triangle.cs
using System;

namespace FigureLibrary
{
    public class Triangle : Figure
    {
        // Конструктор за замовчуванням
        public Triangle() : base() { }

        // Конструктор із 3 параметрами
        public Triangle(double xVal, double yVal, double sideVal)
            : base(xVal, yVal, sideVal)
        {
        }

        // Конструктор копіювання
        public Triangle(Triangle other)
            : base(other)
        {
        }

        // Метод обчислення периметра (для рівностороннього трикутника)
        public double Perimeter()
        {
            return 3.0 * side;
        }

        // Метод обчислення площі
        public double Area()
        {
            return (Math.Sqrt(3.0) / 4.0) * side * side;
        }
    }
}